### ISSUE TITLE
[clang-tidy] `llvmlibc-inline-function-decl` erroneously reports on lambda functions

### ISSUE THREAD

**Issue by jhuber6** [llvm/llvm-project#62147]
The `llvmlibc-inline-function-decl` currently  erroneously reports when defining a lambda function. The reason LLVM's `libc` reports on these variables so that we can ensure functions in headers line up when compiled differently for different TUs. However, the functions emitted from `lambda` values will have internal linkage, so they will be unique to each TU they are implemented in. Here is a reproducer.

```c++
namespace __llvm_libc {

void foo(void (*fn)(void));

void bar() {
  foo([]() {});
}

} // namespace __llvm_libc
```
```console
$ clang-tidy test.h --checks=-'*,llvmlibc*'               
8 warnings generated.
test.h:6:7: warning: '__invoke' must be tagged with the LIBC_INLINE macro; the macro should be placed at the beginning of the declaration [llvmlibc-inline-function-decl]
  foo([]() {});
      ^
test.h:6:7: warning: 'operator void (*)()' must be tagged with the LIBC_INLINE macro; the macro should be placed at the beginning of the declaration [llvmlibc-inline-function-decl]
test.h:6:7: warning: '~(lambda at /home/jhuber/Documents/llvm/llvm-project/build/test.h:6:7)' must be tagged with the LIBC_INLINE macro; the macro should be placed at the beginning of the declaration [llvmlibc-inline-function-decl]
test.h:6:10: warning: 'operator()' must be tagged with the LIBC_INLINE macro; the macro should be placed at the beginning of the declaration [llvmlibc-inline-function-decl]
  foo([]() {});
         ^
Suppressed 4 warnings (4 with check filters).
```
